mne/preprocessing/eog.py

```diff
@@ -49,15 +49,19 @@ def find_eog_events(raw, event_id=998, l_freq=1, h_freq=10,
     eog_inds = _get_eog_channel_index(ch_name, raw)
     logger.info('EOG channel index for this subject is: %s' % eog_inds)
 
-    eog, _ = raw[eog_inds, :]
+    # Reject bad segments.
+    eog, times = raw.get_data(picks=eog_inds, reject_by_annotation='omit',
```

Comment:
So perhaps find the events like in this PR and also do ``reject_by_annotation`` in ``create_eog_epochs``? Also, we probably don't have to do a deprecation cycle, since the effective change is so small (and accounts as a bug fix)?